grouper/src/grouper/edu/internet2/middleware/grouper/plugins/FrameworkStarter.java

```diff
@@ -41,42 +64,54 @@ public static FrameworkStarter getInstance() {
+            LOG.warn("You are setting a value for `grouper.osgi.framework.system.packages.extra`. This generally not needed and should not be used unless there is a good reason to do so");
+            frameworkSystemPackagesExtra.addAll(Arrays.asList(GrouperConfig.retrieveConfig().propertyValueString(GROUPER_OSGI_FRAMEWORK_SYSTEM_PACKAGES_EXTRA, "").split(",")));
+        }
+        if (!frameworkSystemPackagesExtra.isEmpty()) {
+            configMap.put(Constants.FRAMEWORK_SYSTEMPACKAGES_EXTRA, String.join(",", frameworkSystemPackagesExtra));
+        }
 
         try {
             // set up cachedir
             Path cacheDir = Files.createTempDirectory("osgi-cache");
-            String grouperOsgiCacheDir = GrouperConfig.retrieveConfig().propertyValueString("grouper.osgi.cache.rootdir", cacheDir.toString());
+            String grouperOsgiCacheDir = GrouperConfig.retrieveConfig().propertyValueString(GROUPER_OSGI_CACHE_ROOTDIR, cacheDir.toString());
             configMap.put(Constants.FRAMEWORK_STORAGE, grouperOsgiCacheDir);
         } catch (IOException e) {
             throw new RuntimeException("problem with setting up osgi cache directory", e);
         }
 
         // usually, this is a bad idea, but we have several classes that must be loaded from the framework classpath to work,
         // e.g., logging, configuration
-        String packagesForBootDelegationString;
-        if (null != GrouperConfig.retrieveConfig().propertyValueString("grouper.osgi.framework.boot.delegation")) {
-            packagesForBootDelegationString = GrouperConfig.retrieveConfig().propertyValueString("grouper.osgi.framework.boot.delegation");
+        Set<String> packagesForBootDelegation = new HashSet<>();
+        if (null != GrouperConfig.retrieveConfig().propertyValueString(GROUPER_OSGI_FRAMEWORK_BOOT_DELEGATION)) {
+            LOG.warn("You are setting a value for `grouper.osgi.framework.boot.delegation`. This is generally not needed adn should not be used unless there is a good reason to do so");
+            packagesForBootDelegation.addAll(Arrays.asList(GrouperConfig.retrieveConfig().propertyValueString(GROUPER_OSGI_FRAMEWORK_BOOT_DELEGATION).split(",")));
         } else {
-            Set<String> packagesForBootDelegation = new HashSet<>();
-            packagesForBootDelegation.add(LogFactory.class.getPackage().getName());
-            packagesForBootDelegation.add(ConfigPropertiesCascadeBase.class.getPackage().getName());
-            // TODO: why oh why... need to fix this
-            packagesForBootDelegation.add(GrouperExternalSystem.class.getPackage().getName());
-            packagesForBootDelegationString = String.join(",", packagesForBootDelegation);
+            packagesForBootDelegation.add("org.osgi.*");
+            packagesForBootDelegation.add("javax.*");
+            packagesForBootDelegation.add("org.apache.commons.logging");
+            packagesForBootDelegation.add("edu.internet2.middleware.grouper.*");
```

Comment:
need to check this, but could cause a problem if a plugin ends up in the classpath (e.g. class clash)